poster/tests/uniswap_test.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  computeProfitMaximizingTrade,
+  getReserves,
+  swapToPrice
+} from '../src/uniswap';
+import { Whit } from './whit';
+import Web3 from 'web3';
+import {
+  bnToBigInt,
+  maxUint
+} from '../src/util';
+import { toBeNear } from './matcher';
+
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeNear(expected: number, delta: number): CustomMatcherResult
+    }
+  }
+}
+
+expect.extend({toBeNear});
+
+function e(n: number | bigint, exp: number | bigint) {
+  return BigInt(n) * 10n ** BigInt(exp);
+}
+
+describe('computeProfitMaximizingTrade', () => {
+  [
+    {
+      name: "simple",
+      truePriceTokenA: e(100, 6),
+      truePriceTokenB: e(200, 6),
+      reserveA: e(10, 18),
+      reserveB: e(21, 18),
+      expect: {
+        aToB: true,
+        amountIn: 20n
+      }
+    }
+  ].forEach((conf) => {
+    test(conf.name, async () => {
+      let result = computeProfitMaximizingTrade(
+        conf.truePriceTokenA,
+        conf.truePriceTokenB,
+        conf.reserveA,
+        conf.reserveB
+      );
+
+      expect(result).toEqual(conf.expect);
+
+
+    });
+  });
+});
+
+describe('swapToPrice', () => {
+  beforeEach(async () => {
+    jest.setTimeout(300000);
+  });
+
+  [
+    // {
+    //   name: "simple",
+    //   decimalsA: 18n,
+    //   decimalsB: 18n,
+    //   reserveA: e(10, 18),
+    //   reserveB: e(10, 18),
+    //   priceA: e(100, 6),
+    //   priceB: e(200, 6)
+    // },
+    {
+      name: "decimals",
+      decimalsA: 18n,
+      decimalsB: 10n,
+      reserveA: e(10, 18),
+      reserveB: e(10, 10),
+      priceA: e(100, 6),
+      priceB: e(200, 6)
+    }
+  ].forEach(conf => {
+    test.only(conf.name, async () => {
+      let whit = await Whit.init({
```

Comment:
is this a philly cheesesteak reference?

Reply:
It did come up in my head, yes, though also an oblique reference to Walt Whitman maybe?

Reply:
https://en.wikipedia.org/wiki/Walt_Whitman_House did you know walt whitman died just across the river in Camden?